lib/rex/proto/http/client.rb

```diff
@@ -144,11 +144,40 @@ def request_raw(opts = {})
   def request_cgi(opts = {})
```

Comment:
Let's add some tests for this; I think we can just add it directly to 

https://github.com/rapid7/metasploit-framework/blob/c59495cc5f2e788e07a7705427dedefa817ed13f/spec/lib/rex/proto/http/client_spec.rb

We should be able to call the `.to_s` method of the `ClientRequest` response in those tests and verify the HTTP request is as expected